### PR TITLE
LDAP: Make LDAP attribute mapping case-insensitive

### LINKED ISSUE #11300: LDAP config is case sensitive
LDAP config (ldap.toml) is case sensitive in the distinguished names, where it shouldn't be.
This works for me:
```
group_dn = "CN=somegroup,OU=Groups,DC=company,DC=net"
```
But this doesn't:
```
group_dn = "CN=somegroup,OU=Groups,dc=company,dc=net"
```
(with an Active Driectory server if it matters)

Grafana version: 5.0.3 (current latest)
OS: CentOS 7.4

### LINKED ISSUE #58991: LDAP Attributes *might* (must?) be case insensitive
**What happened**:
I typed `samaccountname` in attribute mapping for `username` and it was mapped to a empty string. So only first login created a user.
I was confused, because attribute name was correct, but tried using case-sensitive form just in case, and it worked.

**What you expected to happen**:

Most LDAP tools (everyone that I know of/used) return requested attributes case-insensitively.

Examples:
* openLDAP's ldapsearch
* Python's ldap3 library
* Microsoft's ldp.exe

**How to reproduce it (as minimally and precisely as possible)**:

Configure LDAP with any attribute map with case different from schema's, then go to Grafana LDAP mapping test page e try getting a user. The attribute will be empty.

**Anything else we need to know?**:

Related:
go-ldap/ldap#129
go-ldap/ldap#111
#11300

**Environment**:
- Grafana version: any

I'll propose a fix for this...

### PR DESCRIPTION
**Pull request by markkrj**
Make Grafana's LDAP attribute mapping case-insensitive.
I did a lot of LDAP integrations and never came against this issue, so I think this might be the most common behavior.
Even tools like `ldapsearch` and `ldp.exe` do not care about attribute's letter case.

fix: #11300
fix: #58991

### CODE REVIEW

File: pkg/services/ldap/helpers.go [hunk 1]
```diff
@@ -49,7 +50,8 @@ func getArrayAttribute(name string, entry *ldap.Entry) []string {
 	}
 
 	for _, attr := range entry.Attributes {
-		if attr.Name == name && len(attr.Values) > 0 {
+		if strings.EqualFold(attr.Name, name) && len(attr.Values) > 0 {
+			fmt.Println(len(attr.Values))
```
Comment (mgyongyosi) [inline pkg/services/ldap/helpers.go:54]: Could you please remove this line?

**Review by mgyongyosi — CHANGES_REQUESTED** [review]

File: pkg/services/ldap/helpers.go (hunk 1 above)
Comment (markkrj) [reply pkg/services/ldap/helpers.go]: Done 😅

File: pkg/services/ldap/helpers.go [hunk 2]
```diff
@@ -1,6 +1,7 @@
 package ldap
 
 import (
+	"fmt"
```
Comment (mgyongyosi) [inline pkg/services/ldap/helpers.go:4]:
```suggestion
```

**Review by mgyongyosi — APPROVED** [review]
LGTM. I think it shouldn't cause any problem/breaking change, because the current rule (case-sensitivity) is stricter.

fyi @Jguer

**Review by Jguer — APPROVED** [review]

**Comment by mgyongyosi** [comment]
Thank you @markkrj for this PR! 🙇  It has been merged. 🎉

**Pull request merged**